### PR TITLE
Expand developer instructions

### PR DESCRIPTION
**Pull request by casparvl**
Added some additional instructions on how to use the git-machinery if you want to e.g. test someone else's feature branch.

### CODE REVIEW

**Comment by casparvl** [comment]
Ping @satishskamath for a review :)

**Comment by smoors** [comment]
@casparvl can you first merge the main branch into this PR?

File: README.md [hunk 1]
```diff
@@ -88,17 +88,73 @@ PYTHONPATH=$PYTHONPATH:$EBROOTREFRAME:$eessihome reframe \
...
-This also works on forked repositories, e.g.
+Generally, you'll want to do this from a forked repository though, where someone worked on a feature. E.g.
 
 ```bash
 pip install git+https://github.com/<someuser>/test-suite.git@branchname
 ```
 
-making it an easy way of testing PRs.
+### Check out a feature branch from a fork
+We'll assume you already a local clone of the official test-suite repository, called 'origin'. In that case, executing `git remote -v`, you should see:
```
Comment (satishskamath) [inline README.md:112]: you already **have** a local clone

File: README.md [hunk 2]
```diff
@@ -88,17 +88,73 @@ PYTHONPATH=$PYTHONPATH:$EBROOTREFRAME:$eessihome reframe \
...
 
 ```bash
 pip install git+https://github.com/EESSI/test-suite.git@branchname
 ```
 
-This also works on forked repositories, e.g.
+Generally, you'll want to do this from a forked repository though, where someone worked on a feature. E.g.
 
 ```bash
 pip install git+https://github.com/<someuser>/test-suite.git@branchname
```
Comment (satishskamath) [inline README.md:108]: pip install tested and it works.

**Review by satishskamath — COMMENTED** [review]
So I have explicitly checked the steps and they work properly. The only thing to change is to add a word specified in a comment above.

**Review by satishskamath — APPROVED** [review]

**Pull request merged**